src/Analyser/TypeSpecifier.php

```diff
@@ -2172,6 +2172,29 @@ public function resolveIdentical(Expr\BinaryOp\Identical $expr, Scope $scope, Ty
 			}
 		}
 
+		if ($rightType instanceof IntegerRangeType) {
```

Comment:
inspired by the previous loop with constant-strings, I have added a similar case for integer ranges